static/sw.js

```diff
@@ -0,0 +1,93 @@
+importScripts('/scripts/sw/workbox/workbox-sw.js');
+
+workbox.setConfig({
+    modulePathPrefix: '/scripts/sw/workbox/'
+  });
+workbox.skipWaiting();
+workbox.clientsClaim(); 
+
+workbox.precaching.precacheAndRoute([]);
+
+// cache images
+workbox.routing.registerRoute(
+    /\.(?:png|PNG|gif|GIF|jpg|JPG|jpeg|JPEG|svg|SVG)$/,
+    workbox.strategies.cacheFirst({
+        cacheName: 'images',
+        plugins: [
+            new workbox.cacheableResponse.Plugin({
+                statuses: [0, 200],
+              }),
+            new workbox.expiration.Plugin({
+                maxEntries: 60,
+                maxAgeSeconds: 30 * 24 * 60 * 60, // 30 Days
+            }),
+        ],
+    }),
+);
+
+// cache giphy
+workbox.routing.registerRoute(
```

Comment:
what is your question here? this is caching the image below.

Reply:
I would not start to precache every external image manually within the code. The client will cache it on first view, that's how caching is intended, imo it should not be the devs task to prepare specific stuff.

Reply:
only the route is registered there, as desired the file will be cached on first view only. because of it is an external file it's handled manually here.